### PR TITLE
Make some associated functions of `Color` const

### PR DESCRIPTION
**Pull request by mamekoro**
# Objective
Make the following functions `const` that will be useful to define colors as constants.

- `Color::srgb_from_array`
- `Color::srgba_u8`
- `Color::srgb_u8`

The last two require Rust 1.82.0.

## Solution
- Make them `const`
- Change MSRV to 1.82.0

## Testing
I tested bevy_color only. My machine does not have enough RAM capacity to test the whole bevy.

`cargo test -p bevy_color`

### CODE REVIEW

**Review by ickshonpe — APPROVED** [review]
Don't see any problems.

File: crates/bevy_color/src/color.rs [hunk 1]
```diff
@@ -143,7 +143,7 @@ impl Color {
     /// Creates a new [`Color`] object storing a [`Srgba`] color from [`u8`] values.
     ///
     /// A value of 0 is interpreted as 0.0, and a value of 255 is interpreted as 1.0.
-    pub fn srgba_u8(red: u8, green: u8, blue: u8, alpha: u8) -> Self {
+    pub const fn srgba_u8(red: u8, green: u8, blue: u8, alpha: u8) -> Self {
```
Comment (ickshonpe) [inline crates/bevy_color/src/color.rs:146]: Won't work because of floating arithmetic.

File: crates/bevy_color/src/color.rs [hunk 2]
```diff
@@ -163,7 +163,7 @@ impl Color {
     /// Creates a new [`Color`] object storing a [`Srgba`] color from [`u8`] values with an alpha of 1.0.
     ///
     /// A value of 0 is interpreted as 0.0, and a value of 255 is interpreted as 1.0.
-    pub fn srgb_u8(red: u8, green: u8, blue: u8) -> Self {
+    pub const fn srgb_u8(red: u8, green: u8, blue: u8) -> Self {
         Self::Srgba(Srgba {
             red: red as f32 / 255.0,
```
Comment (ickshonpe) [inline crates/bevy_color/src/color.rs:168]: Same here.

**Review by ickshonpe — CHANGES_REQUESTED** [review]
I think `srgb_from_arrray` can be made const but not the other two.

File: crates/bevy_color/src/color.rs (hunk 1 above)
Comment (alice-i-cecile) [reply crates/bevy_color/src/color.rs:146]: const floating point arithmetic just stabilized :)

Comment (ickshonpe) [reply crates/bevy_color/src/color.rs:146]: ohh I see, I was confused by how the author's bevy_color tests could be passing locally.

Comment (mamekoro) [reply crates/bevy_color/src/color.rs:146]: I had forgotten about Bevy's MSRV.
Should I also change the value of `package.rust-version` in Cargo.toml to `1.82.0` with this pull request?

Comment (alice-i-cecile) [reply crates/bevy_color/src/color.rs:146]: Yes, please do.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**